### PR TITLE
Lowers bag of holding max weight class by one

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Lowers bag of holding maximum size class to the next tier down, preventing storing of Gigantic or Bulky items. Amount of items it can store at once unchanged

## Why It's Good For The Game

Prevents storing of items that possess the Bulky weight class for the purpose of not allowing characters to carry them around unseen. Such as LMGs and laser guns, katanas, hardsuits, other backpacks, dismembered heads.
Prevents nesting backpacks inside bags of holding to bypass the storage limit.
Lessens ability for players to equip themselves for any potential threat on the station by carrying large weapons and spacesuits in their backpack. Bags of holding act more as a printable bigger duffelbag with no slowdown rather than perhaps the most powerful and useful item in the game.

## Changelog
:cl:
balance: Bulky items aren't able to fit inside bags of holding
/:cl:

### CODE REVIEW

**Comment by LivingBatteryBox** [comment]
Booooooo. Trash bag already got nerfed, let people have their fun.

**Comment by ghost** [comment]
> 
> 
> Booooooo. Trash bag already got nerfed, let people have their fun.

They still hold the same amount, this is just so items that are expressly meant to not be hideable in a bag or hoardable don't fit inside

**Comment by francinum** [comment]
> Booooooo. Trash bag already got nerfed, let people have their fun.

Why do people keep saying this.

They weren't nerfed. They were bugged. The bug was fixed.

**Comment by OceanFish1** [comment]
> > Booooooo. Trash bag already got nerfed, let people have their fun.
> 
> Why do people keep saying this.
> 
> They weren't nerfed. They were bugged. The bug was fixed.

Why do people keep saying this? Why do coders say it's not a but it's a  feature?

**Comment by Sinestia** [comment]
I think I'd rather their acquisition be more difficult than this change

**Comment by ghost** [comment]
> 
> I think I'd rather their acquisition be more difficult than this change


I wouldn't mind that either, but I think the reaction would be more negative from people not being able to print bags of holding every round than be unable to fit the armory inside one. It wouldn't much help that anyone who does get their hands on one would still be able to hoard equipment that isn't balanced around them having at all times without anyone else being the wiser

**Comment by bloons3** [comment]
> Booooooo. Trash bag already got nerfed, let people have their fun.

Trash bag of holding was shitcode and never intended.

**Comment by super12pl** [comment]
BoHs already have a number of drawbacks for example teleport interference and high cost.
Also wouldn't decreasing size class disable BoH bombing?

**Comment by RDS88-beeman** [comment]
Obligatory "are you sure you didn't mean to make this nerf PR to the TG repository"

But in all seriousness, isn't the point of a bag of holding to hold, like, really big stuff?
The two influences I think of are that bag from Mary Poppins, which she pulls an entire hatstand out of at one point and the bag of holding from DND, that you can just kidnap whole people with. Bottomless bags are nothing new but ours is already a significant downgrade from the things we're referencing.

While this isn't a movie, and it isn't DND, and balance must be considered, the utility is, I think, balanced by their scarcity and potential danger, given BOH bombs and the teleportation fuckery. There are much more egregious things in the sci lathe alone. 

Maybe it's not balanced to hide a LMG, but who cares when a shotgun with laser scatter shells does more damage faster and if sawn off fits in any bag, and you don't even need to leave science to make an improvised shotgun with them.

> "The most powerful and useful item in the game"

hoooooooo boy. Plant bags fit in a pocket and on belts. Each plant bag holds 50 plants. Each plant has the potential to oneshot anyone it touches and melt ALL of their 
Bio bags also fit in a pocket or on a belt. Each can hold 25 ten use timestop grenades, or smoke grenades filled with whatever stupid stuff you can prepare, or 25 actual grenades, or 25 bottles of whatever chemical deathmix or death virus you can mix.
There are worse things

**Comment by ghost** [comment]
> 
> 
> BoHs already have a number of drawbacks for example teleport interference and high cost.
> Also wouldn't decreasing size class disable BoH bombing?

The only drawback to wearing a BoH every round is that you can't use teleporters/portal guns, and that doesn't cause a problem in 99% of rounds, and if even you do get teleported it's a 50/50 chance you just get teleported somewhere else on the station, *and* you can fit a full hardsuit in your bag of holding for free to make space a non-issue. As for cost, everyone on the station can be equipped with a bag of holding so long as there's one miner onboard. Even if they die, bluespace miners can be constructed to generate infinite resources after the first ore haul.

@RDS88-beeman In DnD you also can't fit things inside a bag of holding if it's too big to fit into the opening of the bag. For SS13 the primary issue is that you can hoard enough equipment into a BoH to have yourself geared for any situation on the station and have mass amounts of weaponry without the downsides those items normally confer, like giving you slowdown in the case of hardsuits, or being visible to everyone who sees you that you're carrying around lethals that can put someone down in 3 seconds. Additionally, you can bypass any balance decisions that declare an item shouldn't be something you can carry around with you all round without anyone being able to see it, arrest you for it, or disarm it off you, such as the Hierophant's staff or armory weapons.

**Comment by ghost** [comment]
Actually yeah this doesn't allow BOH bombing, I'll look into fixing it a few hours from now.

**Comment by utilbird** [comment]
Reducing it to bulky instead of normal would be better imo. The main issue with Bags of Holding is how obtainable they are, this proposed change aims to kill it's demand instead of it's supply, which I really don't like. 

Bags of Holding are supposed to be able to store large items in a small space, that's how almost every other bluespace related storage item works, but that's a lore argument that most people won't care about.

In terms of balance, as mentioned before, bags of holding are too accessible. They're usually one of the first technologies scientists try to obtain and are usually printable by the first or second ore delivery, assuming miners are doing their jobs.

TG addressed this by making bags of holding require a bluespace core to be used, I'm not sure if that's the best possible solution, but it's one to consider.

tl;dr, I would personally rather have strong that are (relatively) hard to obtain than not having strong items.

**Comment by ghost** [comment]
> 
> 
> Reducing it to bulky instead of normal would be better imo. The main issue with Bags of Holding is how obtainable they are, this proposed change aims to kill it's demand instead of it's supply, which I really don't like.
>

To echo what I said on discord, nerfing the obtainability instead of its power would, I think, resulting in the item mostly just being rushed by powergamers or antags who already receive the strongest power boost from this item. People already break into Science to research and print BoHs just for the sake of having one, I can't imagine making it require something like an anomaly core would result in anything other than powergamers carrying around anomaly neutralisers/analyser+signaller all round so they can be the one to get the gamer backpack. And if they do get their hands on one, then the problems inherent to bags of holding don't go away; that person can still hoard all the weapons/enormous items in the world and whip them out to game end an unsuspecting person or bypass a problematic situation someone else created

**Review by PowerfulBacon — DISMISSED** [review]
Good change, Gigantic / Bulky things such as hardsuits and stuff are bulky so you cant carry them the entire round to use just when you need them. If you want them you have to trade for the downsides of the slowness.

**Comment by PowerfulBacon** [comment]
> 
> 
> Reducing it to bulky instead of normal would be better imo. The main issue with Bags of Holding is how obtainable they are, this proposed change aims to kill it's demand instead of it's supply, which I really don't like.
> 
> Bags of Holding are supposed to be able to store large items in a small space, that's how almost every other bluespace related storage item works, but that's a lore argument that most people won't care about.
> 
> In terms of balance, as mentioned before, bags of holding are too accessible. They're usually one of the first technologies scientists try to obtain and are usually printable by the first or second ore delivery, assuming miners are doing their jobs.
> 
> TG addressed this by making bags of holding require a bluespace core to be used, I'm not sure if that's the best possible solution, but it's one to consider.
> 
> tl;dr, I would personally rather have strong that are (relatively) hard to obtain than not having strong items.

Bag of holding use bluespace to store a lot of items in a small space, if the item cant fit in a bag in the first place how is it meant to fit though the hole into the bluespace pocket.

In terms of balance BOHs allow storing of things that shouldn't be able to be stored IE hardsuits, shit tons of energy guns etc. These things are larger for a reasown: Most hardsuits use the slowdown as a downside to prevent powergaming them at all times mechanically, energy guns the same.

I would much rather Bag of holdings remain a useful item (But not one that lets you bypass the majority of balance features) than an item that is incredibly difficult to obtain.

**Comment by WhyIsCaeciliusTaken** [comment]
This is hands down the worst way to nerf the BoH, because it essentially makes it into a glorified dufflebag with the extra downside of being unable to use teleporters. Even TG's nerf is preferable to this nonsense.

**Comment by utilbird** [comment]
> I would much rather Bag of holdings remain a useful item (But not one that lets you bypass the majority of balance features) than an item that is incredibly difficult to obtain.

How is this better though. Now it'd just be a glorified duffelbag, which is even more dumb when you consider it's main downside to the rest of the crew, **it's a bomb.** A very potent, permanent one at that

**Comment by PowerfulBacon** [comment]
> How is this better though. Now it'd just be a glorified duffelbag, which is even more dumb when you consider it's main downside to the rest of the crew, **it's a bomb.** A very potent, permanent one at that

Its a duffelbag with 0 slowdown and significantly more space. The main thing I should mention is isn't GIGANTIC to NORMAL going down 2 weight classes or is BULKY above GIGANTIC?

**Comment by Mat05usz** [comment]
> Even if they die, bluespace miners can be constructed to generate infinite resources after the first ore haul.

Oh boi, did you miss BS miners removal?

Also cringe nerf that looks like was made because one round someone pulled out a lasgun from their BoH and killed you (egun/lasgun change to bulky was also very cringe). In my opinion, what this nerf does is just making life more painful while not adding anything meaningful to the game.

**Comment by Nus127** [comment]
> > How is this better though. Now it'd just be a glorified duffelbag, which is even more dumb when you consider it's main downside to the rest of the crew, **it's a bomb.** A very potent, permanent one at that
> 
> Its a duffelbag with 0 slowdown and significantly more space. The main thing I should mention is isn't GIGANTIC to NORMAL going down 2 weight classes or is BULKY above GIGANTIC?

Heyo, i just came to drop a lil info. The BoH and the Duffelbag inventory sizes arent actually that much different. 
The BoH holds one 50 sheet stack of iron more than a duffelbag, which are normal sized items.

The differences between the two is the slight slowdown of the Duffelbag(which you can just hold in-hand), The w_class that can be accepted inside the bags, one isnt an explosive device, and finally one doesnt have a chance to teleport you from any means into space/ other dangerous places.

**Comment by ghost** [comment]
Bags of holding grant greater than duffelbag space (35 overall item space compared to 31 for duffelbags, **21** for normal backpacks) without the slowdown that's conferred from wearing a duffelbag, or holding one inhand and risking losing it from a slip or someone disarming you. I think it's enough that science is able to print a 1.6x larger backpack to improve storage space for the entire crew. Isn't that enough of a boon for something you can create after clicking research a few times on the RnD console?

**Comment by utilbird** [comment]
Not when you consider the risk they pose to the crew. Bags of holding shouldn't even be common in the first place, with this change, being common is the only aspect that would make them not horrible.

**Comment by PowerfulBacon** [comment]
> Not when you consider the risk they pose to the crew. Bags of holding shouldn't even be common in the first place, with this change, being common is the only aspect that would make them not horrible.

Sounds bad and all but I havent seen a BOH bomb in ages.
It doesnt offer non antags the ability create an accident due to the popup menu and antags can get a BOH if they want just by asking science or breaking in.
Also how would nerfing BOHs make this worse? If anything there would be less BOHs in circulation so this risk would actually be reduced by this PR

**Comment by WhyIsCaeciliusTaken** [comment]
After some thinking I've had a suggestion to change BoHs:

It's a normal backpack with normal storage limits, but with a toggleable bluespace pocket which can hold the larger items as before. Opening/closing the pocket is an ability that takes several seconds and has a visible message. When open, the pocket gives the BoH a different visual sprite and massive slowdown to the user. This allows for the retention of the ability to carry large items around without the issues of "suddenly pulling out six shotguns in a row/hardsuit" etc because of the additional downside of the large items taking a long time to take out of the backpack.

File: code/datums/components/storage/concrete/bag_of_holding.dm [hunk 1]
```diff
@@ -25,3 +25,7 @@
 		qdel(A)
 		return
 	. = ..()
+/datum/component/storage/concrete/bluespace/bag_of_holding/Initialize()
```
Comment (PowerfulBacon) [inline code/datums/components/storage/concrete/bag_of_holding.dm:29]:
```suggestion

/datum/component/storage/concrete/bluespace/bag_of_holding/Initialize()
```

**Comment by ItsTisme** [comment]
I have to disagree on this for the same reasons as described. It makes the BoH only a little better then a duffle bag and makes it a little moot. 

The main utility for me is to carry a  hardsuit as a CE aswel as the other engineering crap and this bassicly nullifies its main upside. This mainly would just remove utility of BoH with carrying heavy items for your job. 

Also the arugment that people stock the entire armory into the bag is something I dont see either unless the armory is wrecked and unsafe to house the weapons.

**Comment by PowerfulBacon** [comment]
Better idea: Make items that apply slowdown transfer their slowdown to the bag.

File: code/datums/components/storage/concrete/bag_of_holding.dm [hunk 2]
```diff
@@ -25,3 +25,8 @@
 		qdel(A)
 		return
 	. = ..()
+
+/datum/component/storage/concrete/bluespace/bag_of_holding/Initialize()
+    . = ..()
+    var/list/exception_cache = typecacheof(list(/obj/item/storage/backpack/holding))
+    exception_hold = exception_cache
```
Comment (utilbird) [inline code/datums/components/storage/concrete/bag_of_holding.dm]:
```suggestion
    var/static/list/exception_cache = typecacheof(list(/obj/item/storage/backpack/holding))
    exception_hold = exception_cache
```

File: code/datums/components/storage/storage.dm [hunk 3]
```diff
@@ -580,7 +581,7 @@
 		if(!stop_messages)
 			to_chat(M, "<span class='warning'>[host] cannot hold [I]!</span>")
 		return FALSE
-	if(I.w_class > max_w_class)
+	if(I.w_class > max_w_class && !is_type_in_typecache(I, exception_hold))
```
Comment (utilbird) [inline code/datums/components/storage/storage.dm]:
```suggestion
	if(I.w_class > max_w_class && !exception_hold[I.type])
```

**Comment by ghost** [comment]
In my mind the main purpose of a bag of holding is to hold large stuff like that. However it not being able to hold backpacks might be good.

**Comment by utilbird** [comment]
the people's republic of beestation has dubbed your PR.. cringe!

**Comment by utilbird** [comment]
in all seriousness, it'd be hard to get a nerf PR like this through because it almost completely invalidates the BoH's place in the game. Which imo, considering how long it's been there for, is a significant factor. The BoH needs some sort of nerf but this is _probably_ the worst way to do it

**Comment by Sinestia** [comment]
This is one case where I think tg's changes are pretty good.

Keeps the BoH as being extremely strong, but means you need a
neutralized anomaly to get one. 

 You could even do without the anomaly cap if you really wanted to, the
point is that there's a bit more effort than clicking print and you're
weighing its value against a phazon. 

 Basically I like it as being an extremely good miracle item instead of just a basic upgrade

**Comment by ghost** [comment]
> 
> 
> in all seriousness, it'd be hard to get a nerf PR like this through because it almost completely invalidates the BoH's place in the game. Which imo, considering how long it's been there for, is a significant factor. The BoH needs some sort of nerf but this is _probably_ the worst way to do it

A PR getting rid of the BoH's storing of Bulky items is going to have to get made eventually, or the problem won't go away. Inventory storage and item size is too important a factor in balancing for an item that bypasses it to exist - even if it were to be made almost impossible to get.

**Comment by ghost** [comment]
A BoH after this PR is a printable 50% upgrade to your storage capacity with the only negative of not being able to use teleportation. It's still excellent for a lathe item.

**Comment by sircope** [comment]
> hoooooooo boy. Plant bags fit in a pocket and on belts. Each plant bag holds 50 plants. Each plant has the potential to oneshot anyone it touches and melt ALL of their

you're right, someone close this and nerf plant bags instead - science needs their big bags

**Comment by Sinestia** [comment]
But it's not a miracle of science anymore. It's a marginal upgrade with
a pretty bad side effect.

 Requiring anomalies as /tg/ did allowed it to be a miracle while being
scarce, and competing with other uses like phazons.

**Comment by ghost** [comment]
> 
> But it's not a miracle of science anymore. It's a marginal upgrade with a pretty bad side effect.  Requiring anomalies as /tg/ did allowed it to be a miracle while being scarce, and competing with other uses like phazons. 
>
A 50% inventory size increase is a huge boon that lets you hoard enough to be prepared for almost any situation. RCD, meds, tools, forcefield projector, mats, weapons, and plenty of space to spare. Making them rarer isn't going to fix that BoHs allow anyone to carry around hidden objects in their backpack that were *meant* to be big and visible to everyone, like maxcaps and shields, and objects that were meant to inconvenience you if you tried having them with you at all times, like hardsuits.

Teleportation is a non-issue for BoHs. I've played with them for about three years and worn one in at least a thousand rounds and I've been screwed by BoH teleportation maybe 5 times in that period, and killed twice. Since the extra space makes it trivial to carry extra meds and tools, it's always easy to escape whatever department or part of maint you got sent to, or to make your way to an airlock if you got sent to space.
